vars/piperPipelineStageMavenStaticCodeChecks.groovy

```diff
@@ -0,0 +1,56 @@
+import com.sap.piper.ConfigurationLoader
+import com.sap.piper.QualityCheck
+import com.sap.piper.ReportAggregator
+
+import static com.sap.piper.Prerequisites.checkScript
+
+def call(Map parameters = [:]) {
+    final String stageName = 'staticCodeChecks'
+    final script = checkScript(this, parameters) ?: null
+
+    piperStageWrapper(stageName: stageName, script: script) {
+
+        String spotBugsIncludeFilterFile = 'default_spotbugs_include_filter.xml'
+        String spotBugsLocalIncludeFilerPath = ".pipeline/${spotBugsIncludeFilterFile}"
+        writeFile file: spotBugsLocalIncludeFilerPath, text: libraryResource(spotBugsIncludeFilterFile)
+
+        String defaultPmdRulesFile = 'default_pmd_rulesets.xml'
+        String pmdRulesPath = ".pipeline/${defaultPmdRulesFile}"
+        writeFile file: pmdRulesPath, text: libraryResource(defaultPmdRulesFile)
+
+        mavenExecuteStaticCodeChecks(script: script,
+            spotBugsIncludeFilterFile: spotBugsLocalIncludeFilerPath,
+            pmdRuleSets: [pmdRulesPath])
+
+        executeWithLockedCurrentBuildResult(
+            script: script,
+            errorStatus: 'FAILURE',
+            errorHandler: script.buildFailureReason.setFailureReason,
+            errorHandlerParameter: 'Maven Static Code Checks',
+            errorMessage: "Please examine the SpotBugs/PMD reports."
+        ) {
+            recordIssues failedTotalHigh: 1,
+                failedTotalNormal: 10,
+                blameDisabled: true,
+                enabledForFailure: true,
+                aggregatingResults: false,
+                tool: spotBugs(pattern: '**/target/spotbugsXml.xml')
+
+            recordIssues failedTotalHigh: 1,
+                failedTotalNormal: 10,
+                blameDisabled: true,
+                enabledForFailure: true,
+                aggregatingResults: false,
+                tool: pmdParser(pattern: '**/target/pmd.xml')
+        }
```

Comment:
Here you recoredIssues as if both checks were configured...

Reply:
done